src/YesSql.Core/Store.cs

```diff
@@ -82,33 +91,66 @@ internal Store(IConfiguration configuration) : this()
             Dialect = Configuration.SqlDialect;
         }
 
-        public async Task InitializeAsync()
+        /// <summary>
+        /// Initializes a <see cref="Store"/> instance using a specific <see cref="Configuration"/> instance.
+        /// </summary>
+        /// <param name="configuration">The <see cref="Configuration"/> instance to use.</param>
+        internal Store(IConfiguration configuration, string[] collections) : this(configuration)
         {
-            IndexCommand.ResetQueryCache();
-            ValidateConfiguration();
+            _collections.AddRange(collections);
+        }
 
-            TypeNames = new TypeService();
+        public async Task InitializeAsync()
+        {
+            if (_isInitialized)
+            {
+                return;
+            }
 
-            if (!string.IsNullOrEmpty(Configuration.Schema))
+            try
             {
-                await using (var connection = Configuration.ConnectionFactory.CreateConnection())
+                await _semaphore.WaitAsync();
```

Comment:
Just before the try block, so that the semaphore is only released if it has been acquired

Reply:
Done